### PR TITLE
feat: add function to import key from file

### PR DESCRIPTION
**Pull request by chiroptical**
This is a WIP PR to implement `importKeyFromFile`. Any feedback is appreciated as I haven't worked this low level before.

### CODE REVIEW

File: src/Crypto/Gpgme/Crypto.hs [hunk 1]
```diff
@@ -452,9 +452,3 @@ verifyInternal ver_op (Ctx {_ctx=ctxPtr}) sig dat = do
     free datBufPtr
 
     return res'
-
-newDataBuffer :: IO (Ptr C'gpgme_data_t)
-newDataBuffer = do
-    resultBufPtr <- malloc
-    checkError "data_new" =<< c'gpgme_data_new resultBufPtr
-    return resultBufPtr
```
Comment (chiroptical) [inline src/Crypto/Gpgme/Crypto.hs:460]: I put this in `Internal` to avoid cyclic dependencies

File: test/real-person/real-person.key [hunk 2]
```diff
@@ -0,0 +1,19 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```
Comment (chiroptical) [inline test/real-person/real-person.key:1]: This is just some random public key I made to attempt importing

Comment (chiroptical) [reply test/real-person/real-person.key:1]: I made is 1024 bytes to try to keep it small

**Comment by chiroptical** [comment]
@rethab I think this is ready for review

File: src/Crypto/Gpgme/Key.hs [hunk 3]
```diff
@@ -61,6 +63,27 @@ getKey (Ctx {_ctx=ctxPtr}) fpr secret = do
         then return . Just $ key
         else return Nothing
 
+-- | Import a key from a file, this happens in two steps: populate a
+-- @gpgme_data_t@ with the contents of the file, import the @gpgme_data_t@
+importKeyFromFile :: Ctx -- ^ context to operate in
+                  -> FilePath -- ^ file path to read from
+                  -> IO (Maybe GpgmeError)
+importKeyFromFile Ctx {_ctx=ctxPtr} fp = do
+  dataPtr <- newDataBuffer
```
Comment (rethab) [inline src/Crypto/Gpgme/Key.hs:72]: I think you need to free this pointer as well

File: test/real-person/real-person.key (hunk 2 above)
Comment (rethab) [reply test/real-person/real-person.key:1]: thanks that's cool 👍

**Review by rethab — COMMENTED** [review]
Thanks for this contribution @chiroptical .

I have merged the default branch into this one and then I noticed that I didn't setup the triggers for action correctly, so I pushed another commit to fix that. I hope that's okay with you.

With regards to your code, it looks great! I believe there's just one place where you might need to add a `free`. I believe there are other examples of using that function in Crypto.hs so perhaps you can take a peek there for inspiration.

**Comment by rethab** [comment]
@chiroptical how's it going? are you still pursuing this?

**Comment by chiroptical** [comment]
Apologies, I totally forgot about this. I'll get it cleaned up soon!

**Comment by chiroptical** [comment]
@rethab `free` is in place and I renamed the tests to follow the new scheme. Good catch!

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -1,6 +1,12 @@
 name: CI
 
-on: push
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
```
Comment (rethab) [inline .github/workflows/ci.yml:9]:
```suggestion
on: push
```

**Review by rethab — APPROVED** [review]
Thanks for this PR @chiroptical . I have simplified the actions workflow -- I hope that's fine with you.

**Comment by rethab** [comment]
> Thanks for this PR @chiroptical . I have simplified the actions workflow -- I hope that's fine with you.

And then I realised that I added that and there was a reason for that 🙈  

well.. it's back now with a comment :)

**Comment by parsonsmatt** [comment]
Any chance a Hackage release could be made for this? 😄

**Comment by rethab** [comment]
hi @parsonsmatt I can't believe I forgot that 😱 I'm just back from vacation, but I'll do it asap :)

**Comment by parsonsmatt** [comment]
Oh, no worries - if I don't get something done immediately it may never be done without reminders 😄

**Comment by rethab** [comment]
Thanks for your patience @parsonsmatt . I have just published 0.6.0.0 :)

**Comment by parsonsmatt** [comment]
Thanks so much!

**Pull request merged**